lib/vector-config-macros/src/configurable.rs

```diff
@@ -350,36 +360,25 @@ fn generate_variant_metadata(
     meta_ident: &Ident,
     variant: &Variant<'_>,
 ) -> proc_macro2::TokenStream {
-    let variant_as_configurable = get_variant_type_as_configurable();
-
     let maybe_title = get_metadata_title(meta_ident, variant.title());
     let description = get_metadata_description(meta_ident, variant.description())
         .expect("enum variants without a description should be rejected during AST parsing");
     let maybe_deprecated = get_metadata_deprecated(meta_ident, variant.deprecated());
     let maybe_custom_attributes = get_metadata_custom_attributes(meta_ident, variant.metadata());
 
+    // We ase a unit tuple here because we don't have access to the actual type of the variant's enum container,
+    // at least not without parsing it specifically as a type token from an amalgamated version of the enum ident... but
+    // it doesn't actually matter because enums don't support default values, which is the only spot in `Metadata<T>`
+    // where the `T` gets used, so we can carry all the other pertinent details with `()`... it's still a littlw wonky, though.
```

Comment:
```suggestion
    // We use a unit tuple here because we don't have access to the actual type of the variant's enum container,
    // at least not without parsing it specifically as a type token from an amalgamated version of the enum ident... but
    // it doesn't actually matter because enums don't support default values, which is the only spot in `Metadata<T>`
    // where the `T` gets used, so we can carry all the other pertinent details with `()`... it's still a little wonky, though.
```

"amalgamated" might be a challenging word that we could sub for something simpler.

Reply:
Fixed in 18c22d4b8.